addons/stock/models/stock_move.py

```diff
@@ -178,12 +179,47 @@ def _default_group_id(self):
     next_serial = fields.Char('First SN')
     next_serial_count = fields.Integer('Number of SN')
     orderpoint_id = fields.Many2one('stock.warehouse.orderpoint', 'Original Reordering Rule', check_company=True)
+    availability_indication = fields.Char('Availability', compute='_compute_availability_indication')
+    availability_state = fields.Selection([
+        ('available', 'Available'),
+        ('waiting', 'Should be available on time'),
+        ('partial', 'Only a part can be reserved'),
+        ('late', 'Will not be available at time'),
+    ], compute='_compute_availability_indication')
 
     @api.onchange('product_id', 'picking_type_id')
     def onchange_product(self):
         if self.product_id:
             self.description_picking = self.product_id._get_description(self.picking_type_id)
 
+    @api.depends('date', 'date_expected', 'product_id.route_ids', 'product_type', 'product_uom_qty', 'reserved_availability')
+    def _compute_availability_indication(self):
+        mto_route = self.env.ref('stock.route_warehouse0_mto')
+        today = date.today()
+        for move in self:
+            if move.product_type != 'product' or move.state == 'assigned':
+                move.availability_state = 'available'
+                precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+                move.availability_indication = float_repr(move.reserved_availability, precision)
+            elif move.state == 'partially_available':
```

Comment:
I have an issue with this comparison. For me, even if the move is partially reserve. It doesn't mean that the remaining is not available. It just means that at last check availabilty, some pieces were missing.